cloudinit/url_helper.py

```diff
@@ -144,6 +144,24 @@ def code(self):
     def __str__(self):
         return self._response.text
 
+    def iter_content(
+        self, chunk_size: Optional[int] = 1, decode_unicode: bool = False
+    ) -> Iterator[bytes]:
+        """Iterates over the response data.
+
+        When stream=True is set on the request, this avoids reading the content
+        at once into memory for large responses.
+
+        :param chunk_size: Number of bytes it should read into memory.
+        :param decode_unicode: If True, content will be decoded using the best
+        available encoding based on the response.
+        """
+        # Provide a fallback encoding in the event the server doesn’t provide
+        # one.
+        if self._response.encoding is None:
+            self._response.encoding = "utf-8"
```

Comment:
Did this encoding issue hit us somewhere? I'm not seeing any cloudinit callsites which pass `decode_unicode=True` so I'm not sure why we need this.

Reply:
Removed